raycicmd/converter.go

```diff
@@ -138,31 +138,80 @@ func (c *converter) convertGroups(gs []*pipelineGroup, filter *stepFilter) (
 		groupNodes = append(groupNodes, groupNode)
 	}
 
-	// Build index and check if we have duplicated user keys.
 	if err := set.buildIndex(); err != nil {
 		return nil, fmt.Errorf("build index: %w", err)
 	}
 
-	// Apply tags filter.
+	// Populate dependsOn.
 	for _, groupNode := range groupNodes {
-		if !filter.hit(groupNode) {
-			continue // group is rejected
+		// A group node is different from a step node.
+		//
+		// When a group is being depended on, all its steps are being depended
+		// on. The gating is equivalent to wait step at the end of the group.
+		//
+		// When a group depensd on a node, it means all its steps depend on the
+		// node. The gating is equivalent to wait step at the beginning of the
+		// group.
+		commonDeps := make(map[string]struct{})
```

Comment:
nit, name it as groupDeps? commonDeps made me though that its scope is more than a group

Reply:
done.